File: knapsack.java
import java.io.IOException;
import java.util.Scanner;

public class q4 {
    public static void main(String args[]) throws IOException
    {
        int i,j=0,max_qty,m,n;
        float sum=0,max;
        Scanner sc = new Scanner(System.in);
        int array[][]=new int[3][20];
        System.out.println("Enter no of items");
        n=sc.nextInt();

        System.out.println("Enter the Selling price of each items");
        for(i=0;i<n;i++)
            array[0][i]=sc.nextInt();

        System.out.println("Enter the profit of each items");
        for(i=0;i<n;i++)
            array[1][i]=sc.nextInt();

        System.out.println("Enter the number of shares of each items");
        for(i=0;i<n;i++)
            array[2][i]=sc.nextInt();

        System.out.println("Enter total liquid cash of knapsack :");
        max_qty=sc.nextInt();

        m=max_qty;
        while(m>=0)
        {
            max=0;
            for(i=0;i<n;i++)
            {
                if(((float)array[1][i])/((float)array[0][i])>max)
                {
                    max=((float)array[1][i])/((float)array[0][i]);
                    j=i;
                }
            }
            if(array[0][j]>m)
            {
                System.out.println("Quantity of item number: " +  (j+1) + " added is " +m);
                sum+=m*max;
                m=-1;
            }
            else
            {
                System.out.println("Quantity of item number: " + (j+1) + " added is " + array[0][j]);
                m-=array[0][j];
                sum+=(float)array[1][j];
                array[1][j]=0;
            }
        }
        System.out.println("The total profit is " + sum);
    }
}

File: C++/Knapsack.cpp
#include<iostream>
#include<vector>
using namespace std;

int knapsack(vector<int> &weight, vector<int> &value, int W) {
    int n = weight.size();
    vector<vector<int>> dp(n + 1, vector<int>(W + 1, 0));
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= W; j++) {
            if (j >= weight[i - 1]) {
                dp[i][j] = max(dp[i - 1][j], dp[i - 1][j - weight[i - 1]] + value[i - 1]);
            } else {
                dp[i][j] = dp[i - 1][j];
            }
        }
    }
    return dp[n][W];
}
int main() {
    int n, W;
    cin >> n >> W;
    vector<int> weight(n), value(n);
    for (int i = 0; i < n; i++) {
        cin >> weight[i] >> value[i];
    }
    cout << knapsack(weight, value, W) << endl;
    return 0;
}
